benchmark/url/whatwg-url-parse.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+const common = require('../common.js');
+const { URL } = require('url');
+const fs = require('fs');
+const path = require('path');
+
+// This benchmark uses 100,000 unique URLs gathered by the top 100 visited
+// websites in the world. It is used to understand the real world impact of
+// the URL parser performance.
+// Reference: https://github.com/ada-url/url-dataset
```

Comment:
This is a rather arbitrary assumption. URLs from a few popular websites might not at all represent typical inputs to Node.js applications.

Reply:
I think one important missing piece is file URLs - which could be very common in Node.js applications (and is probably on a critical path in the module loader). From a glance this seems to be missing from this dataset.

Reply:
> This is a rather arbitrary assumption. URLs from a few popular websites might not at all represent typical inputs to Node.js applications.

I agree; this is why I'm not replacing the existing benchmarks but adding a new one. The current benchmarks are way worse than this assumption because they are the WPT tests and contain 90% failing scenarios focusing on spec compliance. So, the question is, do we want to know how fast can we fail if the input is invalid, or how fast we can process a successful URL? I think we need to measure both, and this is why this PR is needed.

> I think one important missing piece is file URLs

I agree, I'll open a different PR that focuses on it.